yarn-project/aztec-rpc/src/database/database.ts

```diff
@@ -10,6 +10,20 @@ import { NoteSpendingInfoDao } from './note_spending_info_dao.js';
  * addresses, storage slots, and nullifiers.
  */
 export interface Database extends ContractDatabase {
+  /**
+   * Add a eip1271 witness to the database.
+   * @param messageHash - The message hash.
+   * @param witness - An array of field elements representing the eip1271 witness.
+   */
+  addEip1271Witness(messageHash: Fr, witness: Fr[]): Promise<void>;
```

Comment:
I was going to make a sarcastic comment about how important it is that we have mandatory comments, but I've also been guilty of doing this kind of stuff :-P